Updated keyScan to ignore trailing spaces in KEYINFO attributes

dovys opened the pull request:
Hey,
I've recently ran into a rather odd issue where some keys have a trailing space at the end of the card ID:
```
> keyinfo --show-fpr --list --ssh-fpr
S KEYINFO 61D048F46EE1DCE... T D2760001240102010006064000390000  OPENPGP.3 - - MD5:9b:c8:3c:7b:44:... - -
```

This breaks key scanning as the scanner thinks there are 11 parts to it instead of 10.  This change should be safe as gpg-agent returns a dash (-) when the value is actually empty.

Commit:
Updated keyScan to ignore trailing spaces in KEYINFO attributes

I've recently ran into a rather odd issue where some keys have
a trailing space at the end of the card ID:
> keyinfo --show-fpr --list --ssh-fpr
S KEYINFO 61D048F46EE1DCE... T D2760001240102010006064000390000  OPENPGP.3 - - MD5:9b:c8:3c:7b:44:... - -

This breaks key scanning as the scanner thinks there are 11 parts to it
instead of 10.  This change should be safe as gpg-agent returns a dash
when the value is actually empty.

prep commented:
That's odd. However, looking at your code, wouldn't it be simpler to change this line

```go
	parts := strings.Split(line, " ")
```

to this?

```go
	parts := strings.Split(strings.TrimSpace(line), " ")
```

jackkleeman committed:
Add a function to return a list of keygrips indexed by card ID, by solely querying the yubikey, thus ignoring any cached keys.

dovys committed:
Merge pull request #1 from monzo/keygrips

Keygrips function for finding all keys on card

jackkleeman committed:
Trim leading signature zeros if needed

jackkleeman committed:
Merge pull request #2 from monzo/trim-leading-sig-zeros

Trim leading signature zeros if needed

Commit:
Improve KEYPAIRINFO validation, tolerate new format

Newer versions of GPG can add an extra field to the KEYPAIRINFO lines,
for example:

S KEYPAIRINFO F877E5110AE0878B14C75747397DA0755EC51613 OPENPGP.3 sa

vs

S KEYPAIRINFO A6D4D888546D31DA593F65246BB13DB2F98DB42E OPENPGP.3

This change makes our parser tolerate this format. It adds extra
validation on the card ID and keygrip fields, ignoring ones that
are invalid.

I could not run the tests and gave up after five minutes. 🙈
